components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/definitions/AsyncApiParser.java

```diff
@@ -1690,4 +1699,46 @@ public String getAsyncApiDefinitionForStore(API api, String asyncAPIDefinition,
         return Library.writeDocumentToJSONString(aai20Document);
     }
 
+    public String updateAsyncAPIDefinition(String oldDefinition, API apiToUpdate) {
+        Aai20Document document = (Aai20Document) Library.readDocumentFromJSONString(oldDefinition);
+
+        if (document.components == null) {
+            document.components = document.createComponents();
+        }
+
+        // add scopes
+        if (document.components.securitySchemes == null) {
+            document.components.securitySchemes = new HashMap<>();
+        }
+
+        Aai20SecurityScheme oauth2SecurityScheme = new Aai20SecurityScheme(document.components, "oauth2");
+        oauth2SecurityScheme.type = "oauth2";
+
+        if (oauth2SecurityScheme.flows == null) {
+            oauth2SecurityScheme.flows = new Aai20OAuthFlows(oauth2SecurityScheme);
+        }
+        if (oauth2SecurityScheme.flows.implicit == null) {
+            oauth2SecurityScheme.flows.implicit = new Aai20ImplicitOAuthFlow(oauth2SecurityScheme.flows);
+        }
+        oauth2SecurityScheme.flows.implicit.authorizationUrl = "http://localhost:9999";
+        Map<String, String> scopes = new HashMap<>();
+        Map<String, String> scopeBindings = new HashMap<>();
+
+        Iterator<Scope> iterator = apiToUpdate.getScopes().iterator();
+        while (iterator.hasNext()) {
+            Scope scope = iterator.next();
+            scopes.put(scope.getName(), scope.getDescription());
+            scopeBindings.put(scope.getName(), scope.getRoles());
+        }
+        oauth2SecurityScheme.flows.implicit.scopes = scopes;
+
+        Extension xScopeBindings = oauth2SecurityScheme.flows.implicit.createExtension();
+        xScopeBindings.name = "x-scopes-bindings";
```

Comment:
Shall we use constants for `"x-scopes-bindings"`?

Reply:
Fixed